spring-expression/src/main/java/org/springframework/expression/spel/support/ReflectiveMethodResolver.java

```diff
@@ -269,7 +263,13 @@ else if (Proxy.isProxyClass(type)) {
 	 * @since 3.1.1
 	 */
 	protected Method[] getMethods(Class<?> type) {
-		return type.getMethods();
+		Set<Method> methods=new HashSet<>();
+		methods.addAll(Arrays.asList(type.getMethods()));
+		//Add all methods of Object to have methods like toString on Proxy-Objects
+		methods.addAll(Arrays.asList(Object.class.getMethods()));
+
+		Method[] methods1 = methods.toArray(new Method[0]);
+		return methods1;
```

Comment:
I think it would be better to make a change specific to JDK proxies in `getMethods(Class<?>, Object)` instead of having this change globally applied in `getMethods(Class<?>)`.

However, there is no need to modify this PR.

I already have such a change in place locally and will apply that when merging your PR.